File: core-java-modules/core-java-8-datetime/src/main/java/com/baeldung/datetime/UseLocalDate.java
package com.baeldung.datetime;

import java.time.DayOfWeek;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.temporal.ChronoUnit;
import java.time.temporal.TemporalAdjusters;

class UseLocalDate {

    LocalDate getLocalDateUsingFactoryOfMethod(int year, int month, int dayOfMonth) {
        return LocalDate.of(year, month, dayOfMonth);
    }

    LocalDate getLocalDateUsingParseMethod(String representation) {
        return LocalDate.parse(representation);
    }

    LocalDate getLocalDateFromClock() {
        LocalDate localDate = LocalDate.now();
        return localDate;
    }

    LocalDate getNextDay(LocalDate localDate) {
        return localDate.plusDays(1);
    }

    LocalDate getPreviousDay(LocalDate localDate) {
        return localDate.minus(1, ChronoUnit.DAYS);
    }

    DayOfWeek getDayOfWeek(LocalDate localDate) {
        DayOfWeek day = localDate.getDayOfWeek();
        return day;
    }

    LocalDate getFirstDayOfMonth() {
        LocalDate firstDayOfMonth = LocalDate.now()
            .with(TemporalAdjusters.firstDayOfMonth());
        return firstDayOfMonth;
    }

    boolean isLeapYear(LocalDate localDate) {
        return localDate.isLeapYear();
    }

    LocalDateTime getStartOfDay(LocalDate localDate) {
        LocalDateTime startofDay = localDate.atStartOfDay();
        return startofDay;
    }

    LocalDateTime getStartOfDayOfLocalDate(LocalDate localDate) {
        LocalDateTime startofDay = LocalDateTime.of(localDate, LocalTime.MIDNIGHT);
        return startofDay;
    }

    LocalDateTime getStartOfDayAtMinTime(LocalDate localDate) {
        LocalDateTime startofDay = localDate.atTime(LocalTime.MIN);
        return startofDay;
    }

    LocalDateTime getStartOfDayAtMidnightTime(LocalDate localDate) {
        LocalDateTime startofDay = localDate.atTime(LocalTime.MIDNIGHT);
        return startofDay;
    }

    LocalDateTime getEndOfDay(LocalDate localDate) {
        LocalDateTime endOfDay = localDate.atTime(LocalTime.MAX);
        return endOfDay;
    }

    LocalDateTime getEndOfDayFromLocalTime(LocalDate localDate) {
        LocalDateTime endOfDate = LocalTime.MAX.atDate(localDate);
        return endOfDate;
    }

}

File: core-kotlin-modules/core-kotlin-2/src/main/kotlin/com/baeldung/dates/datetime/UseLocalDate.kt
package com.baeldung.dates.datetime

import java.time.DayOfWeek
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.temporal.ChronoUnit
import java.time.temporal.TemporalAdjusters

class UseLocalDate {

    fun getLocalDateUsingFactoryOfMethod(year: Int, month: Int, dayOfMonth: Int): LocalDate {
        return LocalDate.of(year, month, dayOfMonth)
    }

    fun getLocalDateUsingParseMethod(representation: String): LocalDate {
        return LocalDate.parse(representation)
    }

    fun getLocalDateFromClock(): LocalDate {
        return LocalDate.now()
    }

    fun getNextDay(localDate: LocalDate): LocalDate {
        return localDate.plusDays(1)
    }

    fun getPreviousDay(localDate: LocalDate): LocalDate {
        return localDate.minus(1, ChronoUnit.DAYS)
    }

    fun getDayOfWeek(localDate: LocalDate): DayOfWeek {
        return localDate.dayOfWeek
    }

    fun getFirstDayOfMonth(): LocalDate {
        return LocalDate.now().with(TemporalAdjusters.firstDayOfMonth())
    }

    fun getStartOfDay(localDate: LocalDate): LocalDateTime {
        return localDate.atStartOfDay()
    }
}